src/sentry/api/paginator.py

```diff
@@ -353,6 +353,55 @@ def get_result(self, limit, cursor=None, count_hits=False, known_hits=None):
         )
 
 
+class FooOffsetPaginator(object):
```

Comment:
This is a terrible name. :smile: Given that we primarily want a record offset paginator we might be able to accomplish that with a GenericOffsetPaginator.